### DISCUSSION TITLE
Validate YOLOv8 model for different object sizes

### DISCUSSION THREAD

**Discussion by onat-ure** [ultralytics/ultralytics#4944] [Q&A]
Hi, all! I want to validate my trained YOLOv8 model in a way that I will see how the model works for different object sizes. I think that since some instances of a class are too small in the image, their recall and mAP50 results are low. Thus, I want to validate over specific bbox area values (by splitting into 3 as small, medium, and big) to see if this problem stems from object sizes. How can I achieve this?

**Comment by niiish32x** [comment]
you can change the image-size use yolo cmd script
```
imgsz = ....
```

**Reply by glenn-jocher** [reply]
@niiish32x hello! To evaluate your YOLOv8 model's performance on different object sizes, you can indeed analyze the results based on the bounding box areas. While the YOLOv8 repository does not provide a direct command to split the validation into small, medium, and large object sizes, you can post-process the validation results to categorize detections into these size groups.

After running the standard validation process, you can analyze the saved predictions and ground truth annotations. You would categorize each object based on its bounding box area (e.g., small, medium, large) and then calculate the metrics like recall and mAP50 for each category separately.

For more detailed guidance on the validation process, please refer to the Val mode documentation at https://docs.ultralytics.com/modes/val. Remember, this kind of analysis requires some additional scripting on your part to categorize the objects and compute the metrics accordingly. 📊🔍
